### ISSUE TITLE
Incorrect zonedToUtc behavior with remote wall time that doesn't exist in browser timezone.

### ISSUE THREAD

**Issue by hstom** [marnusw/date-fns-tz#45]
For this example have browser TZ in "America/Chicago"

Start with:
start: 2019-11-03 01:00:00 [-10:00] HST | 1572760800000
Add 1 hour:
plusHour: 2019-11-03 01:00:00 [-10:00] HST | 1572764400000
Check start again for mutability:
start: 2019-11-03 01:00:00 [-10:00] HST | 1572760800000
Difference in hours: (should be 1)
dH: 1
Convert both to UTC:
(e.g. zonedTimeToUtc(start, "Pacific/Honolulu")
startUtc: 2019-11-03 05:00:00 [+00:00] UTC | 1572778800000
plusHourUtc: 2019-11-03 05:00:00 [+00:00] UTC | 1572778800000
Difference in hours: (should be 1)
dHUtc: 0

basically the zonedTimeToUtc of a valid timestamp that is equivalent to an invalid timestamp doesn't work

**Comment by hstom** [comment]
Demo that replicates bug if browser timezone is America/Chicago
```import React, { useState } from "react";

import { addHours, differenceInHours } from "date-fns";
import { zonedTimeToUtc, format } from "date-fns-tz";

const blit = date =>
    format(date, "yyyy-MM-dd HH:mm:ss [xxx] z | T", {
        timeZone: "Pacific/Honolulu"
    });

const blitUtc = date =>
    format(date, "yyyy-MM-dd HH:mm:ss [xxx] z | T", {
        timeZone: "UTC"
    });

export default { title: "date-fns hawaii weird" };

const BAD = zonedTimeToUtc("2019-11-02T20:00:00-10:00","Pacific/Honolulu");
const GOOD = zonedTimeToUtc("2019-11-02T21:00:00-10:00","Pacific/Honolulu");

/**
 * with browser in America/Chicago TZ
 * BAD (default) will have a difference when converted to UTC of 0 hours
 */
export const weird = () => {
    const [bad, setBad] = useState(true);

    const notebook = [];

    let i = 0;
    const getKey = () => `i${++i}`;
    const toggle = () => setBad(b => !b);

    const start = bad ? BAD : GOOD;
    notebook.push(<div key={getKey}>
        {bad
            ? <h4>Start with: <span style={{fontWeight: 'bold'}}>BAD</span> <button onClick={toggle}>GOOD</button></h4>
            : <h4>Start with: <button onClick={toggle}>BAD</button> <span style={{fontWeight: 'bold'}}>GOOD</span></h4>
        }
        start: {blit(start)}
    </div>);

    const plusHour = addHours(start, 1);
    notebook.push(<div key={getKey}>
        <h4>Add 1 hour: </h4>
        plusHour: {blit(plusHour)}
    </div>);

    notebook.push(<div key={getKey}>
        <h4>Check start again for mutability: </h4>
        start: {blit(start)}
    </div>);

    const dH = differenceInHours(plusHour, start);

    notebook.push(<div key={getKey}>
        <h4>Difference in hours: (should be 1)</h4>
        dH: {dH}
    </div>);

    const startUtc = zonedTimeToUtc(start, "Pacific/Honolulu");
    const plusHourUtc = zonedTimeToUtc(plusHour, "Pacific/Honolulu");
    notebook.push(<div key={getKey}>
        <h4>Convert both to UTC:
            <div style={{fontSize: '75%'}}>
            (e.g. zonedTimeToUtc(start, "Pacific/Honolulu")
            </div>
        </h4>
        <div>
        startUtc: {blitUtc(startUtc)}
        </div>
        <div>
        plusHourUtc: {blitUtc(plusHourUtc)}
        </div>
    </div>);

    const dHUtc = differenceInHours(plusHourUtc, startUtc);

    notebook.push(<div key={getKey}>
        <h4>Difference in hours: (should be 1)</h4>
        dHUtc: {dHUtc}
    </div>);

    return (<>
        {notebook}
        </>)
};
```

**Comment by scottlamb** [comment]
Unfortunately I think this shows `date-fns-tz`'s core API design concept—that you can use a JavaScript `Date` to represent any date and time that are valid in some remote zone—is flawed. The only solution is to stop using `Date` (and thus `date-fns`).

I like `date-fns` otherwise, and it's a popular library, so please prove me wrong!
